src/main/java/uk/ac/cam/alpha2017/airqualityradar/backend/data/azureconnectors/AzureTableConnector.java

```diff
@@ -38,7 +39,7 @@ private CloudTable getCloudTable(String tableName) throws URISyntaxException, In
         return cloudTable;
     }
 
-    public Iterator<AirDataEntity> getEntitiesBetweenCalendars(String tableName, Calendar fromDate, Calendar toDate) throws StorageException, URISyntaxException, InvalidKeyException, TableDoesNotExistException {
+    public <T extends TableServiceEntity> Iterator<T> getEntitiesBetweenCalendars(String tableName, Calendar fromDate, Calendar toDate, Class<T> classToMapByReflection) throws StorageException, URISyntaxException, InvalidKeyException, TableDoesNotExistException {
```

Comment:
Minor tweak: I would consider renaming to `entityClass`, since the method by which the Azure storage maps the object to the entity class should be independent of this class.